pkg/k8s/fake.go

```diff
@@ -52,15 +55,16 @@ func NewFakeAPIFromManifests(readers []io.Reader) (*KubernetesAPI, error) {
 
 // NewFakeClientSets provides mock Kubernetes ClientSets.
 // TODO: make this private once KubernetesAPI (and NewFakeAPI) supports spClient
-func NewFakeClientSets(configs ...string) (kubernetes.Interface, apiextensionsclient.Interface, spclient.Interface, error) {
+func NewFakeClientSets(configs ...string) (kubernetes.Interface, apiextensionsclient.Interface, spclient.Interface, tsclient.Interface, error) {
```

Comment:
nit: It feels like `NewFakeClientSets`'s return types need to be compressed to a struct or meta interface. This will also help when we add new fake clients down the road.